Crypto/Cipher/ChaChaPoly1305.hs

```diff
@@ -81,6 +81,18 @@ instance ByteArrayAccess Nonce where
   withByteArray (Nonce8  n) = B.withByteArray n
   withByteArray (Nonce12 n) = B.withByteArray n
 
+instance Eq Nonce where
+  (Nonce8  a) == (Nonce8  b) = a == b
+  (Nonce12 a) == (Nonce12 b) = a == b
+  (Nonce8  a) == (Nonce12 b) = a == b
+  (Nonce12 a) == (Nonce8  b) = a == b
+
+instance Ord Nonce where
+  compare (Nonce8  a) (Nonce8  b) = compare a b
+  compare (Nonce12 a) (Nonce12 b) = compare a b
+  compare (Nonce8  a) (Nonce12 b) = compare a b
```

Comment:
Also means that there's no way that a Nonce8 compared to a Nonce12 returns a Eq this way.